specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2022-02-15-preview/cosmos-db.json

```diff
@@ -8263,6 +8507,198 @@
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/ARMResourceProperties"
+        }
+      ],
+      "required": [
+        "properties"
+      ]
+    },
+    "RedistributeThroughputProperties": {
+      "description": "Properties to redistribute throughput for Azure Cosmos DB resource.",
+      "type": "object",
+      "properties": {
+        "resource": {
+          "description": "The standard JSON format of a resource throughput",
+          "$ref": "#/definitions/RedistributeThroughputPropertiesResource"
+        }
+      },
+      "required": [
+        "resource"
+      ]
+    },
+    "RedistributeThroughputPropertiesResource": {
+      "type": "object",
+      "description": "Resource to redistribute throughput for Azure Cosmos DB resource",
+      "properties": {
+        "throughputPolicy": {
+          "description": "ThroughputPolicy to apply for throughput redistribution",
+          "type": "string",
+          "enum": [
+            "none",
+            "equal",
+            "custom"
+          ],
+          "x-ms-enum": {
+            "name": "ThroughputPolicyType",
+            "modelAsString": true
+          }
+        },
+        "targetPhysicalPartitionThroughputInfo": {
+          "type": "array",
+          "description": "Array of PhysicalPartitionThroughputInfoResource objects.",
+          "items": {
+            "$ref": "#/definitions/PhysicalPartitionThroughputInfoResource"
+          }
+        },
+        "sourcePhysicalPartitionThroughputInfo": {
+          "type": "array",
+          "description": "Array of PhysicalPartitionThroughputInfoResource objects.",
+          "items": {
+            "$ref": "#/definitions/PhysicalPartitionThroughputInfoResource"
+          }
+        }
+      },
+      "required": [
+        "throughputPolicy",
+        "targetPhysicalPartitionThroughputInfo",
+        "sourcePhysicalPartitionThroughputInfo"
+      ]
+    },
+    "RetrieveThroughputParameters": {
+      "description": "Cosmos DB retrieve throughput parameters object",
+      "type": "object",
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "description": "Properties to retrieve throughput parameters object",
+          "$ref": "#/definitions/RetrieveThroughputProperties"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/ARMResourceProperties"
+        }
+      ],
+      "required": [
+        "properties"
+      ]
+    },
+    "RetrieveThroughputProperties": {
+      "description": "Properties to retrieve throughput for Azure Cosmos DB resource.",
+      "type": "object",
+      "properties": {
+        "resource": {
+          "description": "The standard JSON format of a resource throughput",
+          "$ref": "#/definitions/RetrieveThroughputPropertiesResource"
+        }
+      },
+      "required": [
+        "resource"
+      ]
+    },
+    "RetrieveThroughputPropertiesResource": {
```

Comment:
>RetrieveThroughputPropertiesResource [](http://example.com/codeflow?start=5&length=36)

Should this type include the throughput numbers as well?

Reply:
This is the retrieve resource so it only needs to contain physical partition id for which throughput is being queried. Response for this query resource will contain the throughput numbers